orchagent/nexthopgroupkey.h

```diff
@@ -33,6 +33,7 @@ class NextHopGroupKey
 
     NextHopGroupKey(const std::string &nexthops, const std::string &weights)
     {
+        m_overlay_nexthops = false;
```

Comment:
>m_overlay_nexthops [](http://example.com/codeflow?start=8&length=18)

Thanks for the fix! I am worried about similar bugs in general. Is it possible to fail build by gcc option, like `-Wuninitialized`?

Reply:
I did some experiment and found `-Weffc++` will detect this type of error, however it is quite verbose.

Reply:
@qiluo-msft interesting, if it catches the error, that's worth considering at some stage of verification..
Just curious, how verbose is it?

Reply:
You can try it and feel the verbose
```diff
diff --git a/configure.ac b/configure.ac
index d7816b6..4fe4c81 100644
--- a/configure.ac
+++ b/configure.ac
@@ -77,6 +77,7 @@ CFLAGS_COMMON+=" -Wstack-protector"
 CFLAGS_COMMON+=" -Wstrict-aliasing=3"
 CFLAGS_COMMON+=" -Wswitch"
 CFLAGS_COMMON+=" -Wswitch-default"
+CFLAGS_COMMON+=" -Wuninitialized"
 CFLAGS_COMMON+=" -Wunreachable-code"
 CFLAGS_COMMON+=" -Wunused"
 CFLAGS_COMMON+=" -Wvariadic-macros"
```